src/Test.EndToEnd.Baselining/Program.cs

```diff
@@ -0,0 +1,123 @@
+    ///  The tests are too slow to run as unit tests and are only applicable when changing the Baselining algorithm.
+    ///  
+    ///  Process
+    ///  =======
+    ///  - When the baselining algorithm is changed, run these tests to check for differences across a wide set of data.
+    ///  - Determine whether the new algorithm appears 'better' overall and investigate specific Results which don't match as they should.
+    ///  - Save the Output logs as the new baseline once you've decided to accept them.
+    /// 
+    ///  
+    ///  Content Folder Structure
+    ///  ========================
+    ///  BaselineE2E\                      [Test Root Path]
+    ///    Input\                          [SARIF log series to test]
+    ///      CloudMine-Spam\               [Series can be organized in any folder structure; using Tool\Version\ContentType\Series here]
+    ///        v1.2.12\              
+    ///          WorkItem\
+    ///            mseng\                  [A folder with files in it is an individual series]
+    ///              Baseline.sarif        [A starting Baseline before any logs are loaded]
+    ///              20191203_1530.sarif   [Each other SARIF log will be baselined in order by name; use DateTime prefixes to order them]
+    ///              20191204_1135.sarif   [...and use the file name suffix to associate them with their source]
+    ///              
+    ///    Output\                         [Each test run will create an Output folder with the results]
+    ///      Summary.log                   [There is a summary log with one line per series showing overall stats]
+    ///      CloudMine-Spam\...            [Output mirrors the Input folder structure]
+    ///        mseng.log                   [Each series has a detail log which shows what happened to each Result in each Log]
+    ///        
+    ///   Expected\                        [If there is an 'Expected' folder, detail logs will be compared between Output and Expected]
+    ///     CloudMine-Spam\...
+    ///        mseng.log                   [Rename 'Output' to 'Expected' to capture the current state as the new baseline]
+    ///        
+    ///   Output_Debug\                    [These are a copy of the 'Output' logs with more than just the Result GUIDs included]
+    ///     CloudMine-Spam\...            
+    ///       mseng.log                    [Change the code in BaseliningDetailEnricher to add the Result properties you need to investigate differences quickly]
+    ///   
+    ///   Expected_Debug\                  [Each run will create an 'enriched' version of the Expected logs also]
+    ///     
+    /// 
+    ///  Execution Steps
+    ///  ===============
+    ///  Running tests will find each series folder in Input and:
+    ///    - Load the initial baseline
+    ///    - Load each log in order
+    ///    - Replace the GUID with a 'RID' (LogIndex + ' ' + ResultIndex)
```

Comment:
>GUID [](start = 25, length = 4)

I assume you are talking about `result.guid`. There are lots of GUID-valued property, so please specify the `objectName.propertyName` if you decide to keep this. But... strictly speaking, that makes the SARIF file invalid. I suggest that instead you add it to the property bag: `result.properties["RID"] = that-thing`.

Reply:
Made comment Result.Guid.

I know I'm misusing Result.Guid. I wanted a simple, readable, stable number to make debugging easy. I need to use the Guid property because that's the only one the Baselining algorithm keeps from both the "baseline" and "current" copies of the result, so I can see what matched up.